### PR TITLE
Stop to migrate and omit < 4.08 support

### PR DESCRIPTION
**Pull request by Nymphium**
This PR migrates from ocaml-migrate-parsetree to ppxlib and removes version migration from the current compiler to 4.04.
All the services has an constraint `ocaml {>= "4.08"}` and generating AST doesn't use new syntactic features, such as let-operator, so there was no reason to support  < 4.08.

### CODE REVIEW

**Comment by Nymphium** [comment]
I found out that we don't need to use ppxlib to just generate AST, so I modified to use only ocaml-compiler-libs to do.

File: src/syntax.ml [hunk 1]
```diff
@@ -182,17 +182,23 @@ let module_ nm vs = Str.module_ (Mb.mk (strloc (Some nm)) (Mod.structure vs))
 (* try body with _ -> with_ *)
 let try_ body with_ = Exp.try_ body [ Exp.case (Pat.any ()) with_ ]
 
+let construct lid p =
+#if OCAML_VERSION >= (4, 13, 0)
+  let p = Option.map (fun p -> ([], p)) p in
```
Comment (Nymphium) [inline src/syntax.ml:187]: Need to pass empty list not to use existensial type variables (<= 4.13.0).

File: aws-gen.opam [hunk 2]
```diff
@@ -20,7 +20,7 @@ depends: [
   "cmdliner" {>= "1.1.0"}
   "fmt"
   "yojson" {>= "1.6.0"}
-  "ocaml-migrate-parsetree" {< "2.0.0"}
+  "ocaml-compiler-libs"
```
Comment (tmcgilchrist) [inline aws-gen.opam:23]: Can you add a lower bound on this? `>= "v0.11.0"` seems sensible via ppxlib.

I have some re-writing passes in the works that need ppxlib.

File: src/util.mli [hunk 3]
```diff
@@ -31,8 +31,8 @@
     POSSIBILITY OF SUCH DAMAGE.
   ----------------------------------------------------------------------------*)
 
-open Migrate_parsetree
-open Ast_404
+(* open Migrate_parsetree *)
+(* open Ast_404 *)
```
Comment (tmcgilchrist) [inline src/util.mli:35]: Just remove these commented out sections.

**Comment by tmcgilchrist** [comment]
Looks good @Nymphium. 

Could you address the 2 minor comments and ensure the commits are split into; 1 changes to the generator to add functionality (either squashing into a single commit or multiple commits if there is a good reason to) and 2 running codegen/fmt via make gen?
Having that split I can review the generator changes more easily.

**Comment by tmcgilchrist** [comment]
https://github.com/tmcgilchrist/ocaml-aws/runs/5828864281

**Review by tmcgilchrist — APPROVED** [review]

**Pull request merged**